### PR TITLE
Use correct BytePos for where when rewriting trait

### LINKED ISSUE #1824: Latest rustfmt panics
```console
$ git clone https://github.com/mit-pdos/distributary.git
$ cd distributary
$ git rev-parse HEAD
ddaf22f4a6e681e687623cc3c99e86cadc7ef86d
$ rustc --version
rustc 1.21.0-nightly (c417ee9ae 2017-07-25)
$ cargo check
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
$ cargo fmt
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: DistinctSources(DistinctSources { begin: ("/home/jon/dev/projects/distributary/src/lib.rs", BytePos(0)), end: ("/home/jon/dev/projects/distributary/src/flow/core/processing.rs", BytePos(100838)) })', /checkout/src/libcore/result.rs:860:4
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:380
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:396
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
   5: std::panicking::begin_panic_new
             at /checkout/src/libstd/panicking.rs:553
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:521
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:497
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
   9: core::result::unwrap_failed
  10: rustfmt_nightly::rewrite::RewriteContext::snippet
  11: rustfmt_nightly::items::rewrite_where_clause
  12: rustfmt_nightly::items::format_trait
  13: rustfmt_nightly::visitor::FmtVisitor::visit_item
  14: rustfmt_nightly::visitor::FmtVisitor::walk_mod_items
  15: rustfmt_nightly::visitor::FmtVisitor::format_separate_mod
  16: rustfmt_nightly::format_input
  17: rustfmt_nightly::run
  18: rustfmt::execute
  19: rustfmt::main
  20: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
  21: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:458
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:59
  22: __libc_start_main
  23: _start
```

### LINKED ISSUE #1833: Weird code duplication
I haven't double-checked for duplicates, so I apologize in advance if this has already been reported:

```
$ rustup run nightly rustc --version
rustc 1.20.0-nightly (598eddf4f 2017-07-24)
$ rustup run nightly rustfmt --version
0.2.1-nightly ( )
```

**Input**:
```rust
trait Y where Self: Sized {
}
```
**Output**:
```rust
trait Y
trait Y
where
    Self: Sized
{
}
```

**EDIT**: updated to simpler example.
**EDIT 2**: added compiler version.

### PR DESCRIPTION
**Pull request by topecongiro**
Closes #1824.
Closes #1833.

### CODE REVIEW

File: tests/target/issue-1824.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+use std::collections::{HashMap, HashSet};
```
Comment (nrc) [inline tests/target/issue-1824.rs]: It would be better I think to have a minimal test case for this issue - can you cut this down to a few lines?

**Comment by topecongiro** [comment]
Thanks for feedback!
Rebased & fixed a typo.

**Comment by topecongiro** [comment]
The current behavior against the test:
```diff
+pub trait Ingredient
pub trait Ingredient
where
    Self: Send,
{
} 
```

**Comment by nrc** [comment]
Perfect, thank you!

**Pull request merged**